training/src/main/java/wolox/training/models/Book.java

```diff
@@ -0,0 +1,126 @@
+package wolox.training.models;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import org.aspectj.lang.annotation.RequiredTypes;
+
+@Entity
+public class Book {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    public Book() {
+        this.id = null;
+    }
+
+
+    public Book(Long id, String author, String image, String title, String subtitle, String publisher,
+        String year, Integer pages, String isbn) {
+        this.id = id;
```

Comment:
Use the setters, so that in the future you can isolate any asserts you may have regarding the data only in said setters :)

Reply:
So this constructor is unnecesary, right?

Reply:
I think what Lula wanted to say is that you should use the setters inside the constructor. This allows you to isolate the setting process. This is useful for various reasons: 
- in the future you might need to make some sort of validation or add logic to the setting methods, e.g. maybe you need to validate that the isbn contains certain characters. Any code that you add to the setter, you would have to add to the constructor too. You can see why generating duplicate code is inconvenient to maintain. 
- if you don't use the setter method, you will need to test the Constructor by itself. 

Regarding your question, you probably won't use this constructor anyway, but it doesn't hurt to keep it :)

Reply:
EXACTLY what Jime said :)